delta-storage/src/main/java/io/cdap/delta/store/DraftService.java

```diff
@@ -133,15 +151,142 @@ public TableList listDraftTables(DraftId draftId, Configurer configurer) throws
+   * instantiated in order to generate this list. This is an expensive operation.
+   *
+   * If the plugin cannot be instantiated due to missing draft information,
+   * an {@link InvalidDraftException} will be thrown.
+   *
+   * @param draftId id of the draft
+   * @param configurer configurer used to instantiate plugins
+   * @param db the database the table lives in
+   * @param table the table to assess
+   * @return list of tables readable by the source in the draft
+   * @throws DraftNotFoundException if the draft does not exist
+   * @throws InvalidDraftException if the table list cannot be fetched because the draft is invalid
+   * @throws TableNotFoundException if the table does not exist
+   * @throws IOException if the table detail could not be read
+   */
+  public TableAssessment assessTable(DraftId draftId, Configurer configurer, String db, String table)
+    throws IOException, TableNotFoundException {
+    Draft draft = getDraft(draftId);
+    DeltaConfig deltaConfig = draft.getConfig();
+    TableRegistry tableRegistry = createTableRegistry(draft, configurer);
+    TableAssessor<List<ColumnDetail>> sourceTableAssessor = createTableAssessor(configurer, deltaConfig.getSource());
+    TableAssessor<Schema> targetTableAssessor = createTableAssessor(configurer, deltaConfig.getTarget());
+    return assessTable(new SourceTable(db, table), tableRegistry, sourceTableAssessor, targetTableAssessor);
+  }
+
+  /**
+   * Assess the entire pipeline based on the tables the source will read and the capabilities of the target.
+   * An instance of the target plugin will be instantiated in order to generate this list.
+   * This is an expensive operation.
+   *
+   * If the plugin cannot be instantiated due to missing draft information,
+   * an {@link InvalidDraftException} will be thrown.
+   *
+   * @param draftId id of the draft
+   * @param configurer configurer used to instantiate plugins
+   * @return list of tables readable by the source in the draft
+   * @throws DraftNotFoundException if the draft does not exist
+   * @throws InvalidDraftException if the table list cannot be fetched because the draft is invalid
+   */
+  public PipelineAssessment assessPipeline(DraftId draftId, Configurer configurer) {
     Draft draft = getDraft(draftId);
+    DeltaConfig deltaConfig = draft.getConfig();
+    deltaConfig.validatePipeline();
+
+    TableRegistry tableRegistry = createTableRegistry(draft, configurer);
+    TableAssessor<List<ColumnDetail>> sourceTableAssessor = createTableAssessor(configurer, deltaConfig.getSource());
+    TableAssessor<Schema> targetTableAssessor = createTableAssessor(configurer, deltaConfig.getTarget());
+
+    List<Problem> connectivityIssues = new ArrayList<>();
+    List<TableSummaryAssessment> tableAssessments = new ArrayList<>();
+    // go through all tables that the pipeline should read, fetching detail about each of the tables
+    for (SourceTable sourceTable : draft.getConfig().getTables()) {
```

Comment:
Does this mean the accessment can only be done if the draft provides the specifc table names? What will happen if the draft does not specify it and wants to read from all tables?

Reply:
for the first cut, there is no way to tell it to read all tables in a database. The current mocks force the user to select tables.
That being said, it's pretty straightforward to change it such that specifying nothing means reading everything, so I'll modify that.